sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentAsync.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.formrecognizer;
+
+import com.azure.ai.formrecognizer.models.FormPage;
+import com.azure.ai.formrecognizer.models.OperationResult;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.polling.PollerFlux;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sample for extracting layout information using input stream.
+ */
+public class RecognizeContentAsync {
+
+    /**
+     * Main method to invoke this demo.
+     *
+     * @param args Unused. Arguments to the program.
+     */
+    public static void main(final String[] args)  {
+        // Instantiate a client that will be used to call the service.
+
+        FormRecognizerAsyncClient client = new FormRecognizerClientBuilder()
+            .apiKey(new AzureKeyCredential("{api_key}"))
+            .endpoint("https://{endpoint}.cognitiveservices.azure.com/")
+            .buildAsyncClient();
+
+        PollerFlux<OperationResult, IterableStream<FormPage>> recognizeLayoutPoller =
+            client.beginRecognizeContentFromUrl("file_source_url");
```

Comment:
I know we said Java might diverge on this (doing file for sync, url for async), but I think that because you've added a recognizeReceipts and a recognizeReceipts FromURL, you can have the other sync and async functions call the files functions